src/Admin/FieldDescriptionInterface.php

```diff
@@ -20,87 +20,79 @@ interface FieldDescriptionInterface
 {
     /**
      * set the field name.
-     *
-     * @param string $fieldName
      */
-    public function setFieldName($fieldName);
+    public function setFieldName(?string $fieldName): void;
 
     /**
      * Returns the field name.
      *
      * @return string the field name
      */
-    public function getFieldName();
+    public function getFieldName(): ?string;
 
     /**
      * Set the name.
-     *
-     * @param string $name
      */
-    public function setName($name);
+    public function setName(?string $name): void;
 
     /**
      * Returns the name, the name can be used as a form label or table header.
      *
      * @return string the name
      */
-    public function getName();
+    public function getName(): ?string;
 
     /**
      * Returns the value represented by the provided name.
      *
-     * @param string     $name
      * @param mixed|null $default
      *
      * @return mixed the value represented by the provided name
      */
-    public function getOption($name, $default = null);
+    public function getOption(string $name, $default = null);
 
     /**
      * Define an option, an option is has a name and a value.
      *
-     * @param string $name
-     * @param mixed  $value
+     * @param mixed $value
      */
-    public function setOption($name, $value);
+    public function setOption(string $name, $value): void;
 
     /**
      * Define the options value, if the options array contains the reserved keywords
      *   - type
      *   - template.
      *
-     * Then the value are copied across to the related property value
+     * Then the value are copied across to the related property value.
+     *
+     * @param array<string, mixed> $options
      */
-    public function setOptions(array $options);
+    public function setOptions(array $options): void;
 
     /**
      * Returns options.
      *
-     * @return array options
+     * @return array<string, mixed>
      */
-    public function getOptions();
+    public function getOptions(): array;
 
     /**
-     * Returns the template used to render the field.
-     *
-     * @param string $template
+     * Sets the template used to render the field.
      */
-    public function setTemplate($template);
+    public function setTemplate(string $template): void;
 
     /**
      * Returns the template name.
      *
      * @return string|null the template name
      */
-    public function getTemplate();
+    public function getTemplate(): ?string;
 
     /**
-     * Returns the field type, the type is a mandatory field as it used to select the correct template
+     * Sets the field type. The type is a mandatory field as it's used to select the correct template
      * or the logic associated to the current FieldDescription object.
-     *
-     * @param string $type
      */
-    public function setType($type);
+    public function setType(?string $type): void;
```

Comment:
There is something weird `getType` return int|string, but setType has a `string` param.